pkg/version/version_info.go

```diff
@@ -0,0 +1,68 @@
+package version
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+type versionInfo struct {
+	major   int
+	minor   int
+	release int
+}
+
+// compareVersionInfo returns:
+// 	0: if a == b
+//  n > 0: if a > b
+//  n < 0: if a < b
+//  0 with error: if a == nil || b == nil
+func compareVersionInfo(a *versionInfo, b *versionInfo) (int, error) {
```

Comment:
I'd suggest to remove the pointers, and just assume they are correct and leave the error verification problem to an upper level when it can actually be wrong.

Reply:
pointers here are used to avoid pass-by-value due to higher memory consumption

Reply:
These are supposed to be executed every 5 minutes, so that's not really a problem.